resolver/src/lookup_ip.rs

```diff
@@ -149,10 +156,20 @@ impl<C: ClientHandle + 'static> Future for InnerLookupIpFuture<C> {
 
 /// returns a new future for lookup
 fn strategic_lookup<C: ClientHandle + 'static>(
+    host: &str, 
     name: Name,
     strategy: LookupIpStrategy,
     client: CachingClient<C>,
 ) -> Box<Future<Item = Lookup, Error = io::Error>> {
+    if let Some(addrs) = lookup_static_host(host) {
+        let addrs = addrs.into_iter().map(|a| match a {
+            IpAddr::V4(ip4) => RData::A(ip4),
```

Comment:
what about just storing RData in the map, this would save the translation here...